numpy/_typing/_char_codes.py

```diff
@@ -23,14 +23,16 @@
 _ShortCodes = Literal["short", "h", "=h", "<h", ">h"]
 _IntCCodes = Literal["intc", "i", "=i", "<i", ">i"]
 _IntPCodes = Literal["intp", "p", "=p", "<p", ">p"]
-_IntCodes = Literal["long", "int", "int_", "l", "=l", "<l", ">l"]
+_IntCodes = Literal["int", "int_", "l", "=l", "<l", ">l"]
+_LongCodes = Literal["long"]
 _LongLongCodes = Literal["longlong", "q", "=q", "<q", ">q"]
 
 _UByteCodes = Literal["ubyte", "B", "=B", "<B", ">B"]
 _UShortCodes = Literal["ushort", "H", "=H", "<H", ">H"]
 _UIntCCodes = Literal["uintc", "I", "=I", "<I", ">I"]
 _UIntPCodes = Literal["uintp", "P", "=P", "<P", ">P"]
-_UIntCodes = Literal["ulong", "uint", "L", "=L", "<L", ">L"]
+_UIntCodes = Literal["uint", "L", "=L", "<L", ">L"]
+_ULongCodes = Literal["ulong"]
```

Comment:
```suggestion
_ULongCodes = Literal["ulong", "uint", "L", "=L", "<L", ">L"]
```

Reply:
These should be considered as aliases for the same thing, so I don't see why we'd want to give them a different typecode.

Reply:
There is a tricky thing with the follow up, but right now, `ulong` *is* a correct type code for `uint`, so renaming it doesn't really matter.  Maybe it actually makes sense to make: `_UIntCodes = _ULongCodes`?

What I am unclear about is what needs to change if we change this to `_UIntCodes = _UIntpCodes` later.  @BvB93 gave a diff in the PR doing that, but it is not quite clear to me that it is complete now.

Reply:
I think as `np.uint` will in the future diverge from `np.ulong` (in the direction of `np.uintp`) then I would keep `_UIntCodes` and `_ULongCodes` separately (same applies to `np.int_`)

Reply:
I updated it so now:
```python
_ULongCodes = Literal["ulong", "uint", "L", "=L", "<L", ">L"]
_UIntCodes = _ULongCodes
```